api/src/scripts/configure_preset_dashboards.ts

```diff
@@ -0,0 +1,55 @@
+import { existsSync, mkdirSync, readdirSync, readFileSync } from 'fs';
+import path from 'path';
+import { Like } from 'typeorm';
+import { dashboardDataSource } from '../data_sources/dashboard';
+import Dashboard from '../models/dashboard';
+import { PRESET_PREFIX } from '../utils/constants';
+
+async function upsert() {
+  console.info('Starting upsert of preset dashboards');
+  if (!dashboardDataSource.isInitialized) {
+    await dashboardDataSource.initialize();
+  }
+  const queryRunner = dashboardDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+  try {
+    const dashboardRepo = queryRunner.manager.getRepository(Dashboard);
+
+    const basePath = path.join(__dirname, '../preset_dashboards');
+    if (!existsSync(basePath)) {
+      console.error('folder for preset dashboards not found. Please run the script for getting preset dashboards first');
+      process.exit(1);
+    }
+    const files = readdirSync(basePath);
+    const dashboardNames = files.map((file) => { return `${PRESET_PREFIX}${file.substring(0, file.lastIndexOf('.'))}` });
+    const currentPresetDashboards = await dashboardRepo.findBy({ name: Like(`${PRESET_PREFIX}%`) });
+
+    const removed = currentPresetDashboards.filter((dashboard) => { return !dashboardNames.includes(dashboard.name) });
+    if (removed.length) {
+      await dashboardRepo.delete(removed.map((r) => { return r.id }));
+    }
+
+    for (let i = 0; i < files.length; i++) {
+      const config: Record<string, any> = JSON.parse(readFileSync(path.join(basePath, files[i]), 'utf-8'));
+      const name = dashboardNames[i];
+      let dashboard = await dashboardRepo.findOneBy({ name });
+      if (!dashboard) {
+        dashboard = new Dashboard();
+        dashboard.name = name;
+      }
+      dashboard.content = config;
+      await dashboardRepo.save(dashboard);
```

Comment:
According to our previous discussion, a preset dashboard should use the file name as its id.

In some scenarios, the id(or name) must be hard-coded in the frontend code. So we need to know the dashboard's id before inserting it into the database.

Reply:
The ID is an auto-generated UUID so this is not possible. Is it possible to use the name instead?

Reply:
In order to fetch the preset dashboard using the name, 2 things may need to be done:

1. add an API to fetch dashboard details by name
2. add a constraint to make sure that the preset dashboard name is unique

Reply:
1. This already exists. /dashboard/list
2. All dashboard names are unique. This is a constraint on the database level.

Reply:
1. Or do you mean that there needs to be a getByName interface?

Reply:
> 1. Or do you mean that there needs to be a getByName interface?

Yes

Reply:
Okay I will add it

Reply:
Done